log/tracing_logger.go

```diff
@@ -0,0 +1,55 @@
+package log
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// NewTracingLogger enables tracing by preformatting all errors (if they
+// implement stackTracer interface) using "%+v" verb.
+//
+// All errors returned by https://github.com/pkg/errors implement this
+// interface.
+//
+// For debugging purposes only as it doubles the amount of allocations.
+func NewTracingLogger(next Logger) Logger {
+	return &tracingLogger{
+		next: next,
+	}
+}
+
+type stackTracer interface {
+	StackTrace() errors.StackTrace
+}
+
+type tracingLogger struct {
+	next Logger
+}
+
+func (l *tracingLogger) Info(msg string, keyvals ...interface{}) error {
+	return l.next.Info(msg, formatErrors(keyvals)...)
+}
+
+func (l *tracingLogger) Debug(msg string, keyvals ...interface{}) error {
+	return l.next.Debug(msg, formatErrors(keyvals)...)
+}
+
+func (l *tracingLogger) Error(msg string, keyvals ...interface{}) error {
+	return l.next.Error(msg, formatErrors(keyvals)...)
+}
+
+func (l *tracingLogger) With(keyvals ...interface{}) Logger {
+	return &tracingLogger{next: l.next.With(formatErrors(keyvals)...)}
+}
+
+func formatErrors(keyvals []interface{}) []interface{} {
+	newKeyvals := make([]interface{}, len(keyvals))
+	copy(newKeyvals, keyvals)
+	for i := 0; i < len(newKeyvals)-1; i += 2 {
+		if err, ok := newKeyvals[i+1].(stackTracer); ok {
+			newKeyvals[i+1] = fmt.Sprintf("%+v", err)
```

Comment:
It is actually nice to leave the original error in some cases... but the printout should be as you test.

I would wrap it with another error that prints stuff out. Will make another commit you can accept or reject.

Why?  Some handlers might want access to the error itself and the StackTrace, not just in text form.